### PR TITLE
Development

### PR DESCRIPTION
**Pull request by nikitakirad**
Completed the assignment

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "blog-assignment",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@stardust-ui/docs-components": "^0.40.0",
+    "@testing-library/jest-dom": "^4.2.4",
```
Comment (chetanbura) [inline package.json]: Remove unused packages from package.json

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    <Switch>
+      <Route path="/login"  render={props => <Auth {...props} />} />
+      <Route path="/"  component={Blog} />
+    </Switch>);
+
+    if(this.props.isAuthenticated){
+      routes=(
+      <Switch>
+        <Route path="/login"  render={props => <Auth {...props} />} />
+        <Route path="/logout" render={() =><Logout/>} />
```
Comment (chetanbura) [inline src/App.js]: convert to dynamic routes here using map.

File: src/axios-orders.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+const instance = axios.create({
+    baseURL: 'https://blog-app-e7a85.firebaseio.com/'
```
Comment (chetanbura) [inline src/axios-orders.js]: look at filename `axios-orders.js`
do not copy-paste files from other projects.
or you can give a file name appropriate as per the functionality.

File: src/containers/Blog/Blog.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            .then( response => {
+                const fetchedPostsFromServer = [];
+                // console.log(res.data);
+                for ( let key in response.data ) {
+                    fetchedPostsFromServer.push( {
+                        ...response.data[key],
+                        id: key
+                    } );
+                    this.setState({fetchedposts:fetchedPostsFromServer});
+                    // console.log(fetchedOrders);
```
Comment (chetanbura) [inline src/containers/Blog/Blog.js]: remove `console.logs` from all files.

**Review by chetanbura — COMMENTED** [review]
Please have a look on comments and start resolving.

File: src/components/AllInfoOfParticularPost/AllInfoOfParticularPost.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import {Button,TextArea,Modal,Input} from 'semantic-ui-react';
+const allinfoofparticularpost=(props)=>{
+    return(
+            <>
+                <Modal trigger={<Button basic color='blue'>More details...</Button>}>
+                    <Modal.Header>Blog</Modal.Header>
+                    <Modal.Content>
+                    <Modal.Description>
+                        <label>TITLE:</label>
+                        {/* <Input name="title" onChange={(event)=>props.onEditInput(props.id,event)} value={props.title}/><br></br><br></br> */}
```
Comment (chetanbura) [inline src/components/AllInfoOfParticularPost/AllInfoOfParticularPost.js]: remove unwanted code or commented code.

File: src/containers/Auth/Auth.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                <Segment textAlign="center" size="big">
+                    {errorMessage}
+                    {validationmsg}
+                    <Header>Enter the login Details</Header>
+                    <form onSubmit={this.submitHandler}>
+                
+                    <label>Email:</label><br></br>
+                    <Input name="email" onChange={this.inputChangedHandler} /><br></br><br></br>
+                    <label>Password:</label><br></br>
+                    <Input name="password" onChange={this.inputChangedHandler} /><br></br><br></br>
```
Comment (chetanbura) [inline src/containers/Auth/Auth.js]: For password input, it should be `type=password` not `type=text`

File: src/components/Editpost/Editpost.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    }
+    savedUpdatedDataToServerHandler=()=>{
+        let valid=false;
+        valid=Validation(this.state.loadedData);
+        console.log(valid);
+        if(valid){
+        Axios.patch(`posts/${this.props.match.params.id}.json`,this.state.loadedData)
+    .then( response => {
+         console.log( response.data );
+        alert("Successfully updated...");
```
Comment (chetanbura) [inline src/components/Editpost/Editpost.js:28]: After successfully update of `post` it should be redirected to the `post` list page.

File: src/components/Newpost/Newpost.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+                    <Modal.Description>
+                        <label><span style={{color: "red"}}>*</span>TITLE:</label><br></br>
+                        <Input name="title" onChange={props.onInputChange} /><br></br><br></br>
+                        <label><span style={{color: "red"}} >*</span>CONTENT:</label><br></br>
+                        <TextArea rows="4" onChange={props.onInputChange} name="content" /><br></br><br></br>
+                        <label><span style={{color: "red"}}>*</span>AUTHOR:</label><br></br>
+                        <Input name="author" onChange={props.onInputChange} /><br></br><br></br>
+                        <label>AGE:</label><br></br>
+                        <Input name="age" onChange={props.onInputChange} /><br></br><br></br>
+                        <Button color="grey" onClick={props.savedDataToServer}>Add</Button>
```
Comment (chetanbura) [inline src/components/Newpost/Newpost.js:18]: After creating new `Post` it should be taken user to `Post` listing page.

File: src/App.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+import Blog from './containers/Blog/Blog';
+import {connect} from 'react-redux';
+const Auth = lazy(() => import('./containers/Auth/Auth'));
+const Logout = lazy(()=>import('./containers/Auth/Logout/Logout'));
+const Editpost = lazy(()=>import('./components/Editpost/Editpost'));
+
+class App extends Component {
+  
+  render(){
+    let routes=null;
```
Comment (chetanbura) [inline src/App.js]: On refresh user session gets over can you fix this.

File: package.json (hunk 1 above)
Comment (nikitakirad) [reply package.json]: Sir,I have removed all unused packages from package.json

File: src/App.js (hunk 2 above)
Comment (nikitakirad) [reply src/App.js]: Sir,I have converted it to dynamic routes using map

File: src/axios-orders.js (hunk 3 above)
Comment (nikitakirad) [reply src/axios-orders.js]: sir, i have changed the file name

File: src/containers/Blog/Blog.js (hunk 4 above)
Comment (nikitakirad) [reply src/containers/Blog/Blog.js]: Sir,I have removed console.logs

File: src/components/AllInfoOfParticularPost/AllInfoOfParticularPost.js (hunk 5 above)
Comment (nikitakirad) [reply src/components/AllInfoOfParticularPost/AllInfoOfParticularPost.js]: Sir,I have removed the unwanted commented code.

File: src/containers/Auth/Auth.js (hunk 6 above)
Comment (nikitakirad) [reply src/containers/Auth/Auth.js]: Sir,Changed the input type

File: src/components/Editpost/Editpost.js (hunk 7 above)
Comment (nikitakirad) [reply src/components/Editpost/Editpost.js:28]: SIr,done with this part also

File: src/components/Newpost/Newpost.js (hunk 8 above)
Comment (nikitakirad) [reply src/components/Newpost/Newpost.js:18]: sir,it is taking to post listing page

File: src/App.js (hunk 9 above)
Comment (nikitakirad) [reply src/App.js]: Sir,done with this functionality

File: src/App.js [hunk 10]
```diff
@@ -1,36 +1,44 @@
 import React, { Component,lazy,Suspense } from 'react';
-import {withRouter,Route,Switch} from 'react-router-dom'
+import {withRouter,Route} from 'react-router-dom'
 import Blog from './containers/Blog/Blog';
 import {connect} from 'react-redux';
+import * as actions from './store/actions/index';
 const Auth = lazy(() => import('./containers/Auth/Auth'));
 const Logout = lazy(()=>import('./containers/Auth/Logout/Logout'));
```
Comment (chetanbura) [inline src/App.js]: import this lazy in the map not separately by pasing `path` as paramerte.

**Review by chetanbura — COMMENTED** [review]
Added few review comments

File: src/App.js (hunk 10 above)
Comment (nikitakirad) [reply src/App.js]: Sir,I have added lazy in the map

**Pull request merged**